spark3/main.py

```diff
@@ -14,15 +15,28 @@ class Spark3:
 
     def __init__(self, spark: SparkSession,
                  trace: Optional[DataFrame] = None,
-                 log: Optional[DataFrame] = None):
+                 log: Optional[DataFrame] = None,
+                 trace_condition: Optional[Conditions] = None,
+                 log_condition: Optional[Conditions] = None):
         """
         :param spark: :class:`SparkSession`
         :param trace: :class:`DataFrame` to store original data to decode contract functions
         :param log: :class:`DataFrame` to store original data to decode contract events
         """
+        if trace is None and log is None:
+            raise ValueError('Either trace or log should be provided')
+
+        if trace is not None and trace_condition is None:
```

Comment:
This requirements maybe is too strict and make this client hard to use

Reply:
move these asserts to the specific method

Reply:
> This requirements maybe is too strict and make this client hard to use

move these asserts to the specific method